Competition2017/src/competition/subsystems/fuel_launcher/FuelLauncherSubsystem.java

```diff
@@ -0,0 +1,185 @@
+        
+        slave.setControlMode(TalonControlMode.Follower);
+        slave.set(master.getDeviceID());*/
+    }
+    
+    private void updateMotorConfig(XCANTalon motor) {
+        motor.setP(p.get());
+        motor.setI(i.get());
+        motor.setD(d.get());
+        motor.setF(f.get());
+    }
+    
+    /**
+     * Sets the output power of the shooter directly - no PID of any kind
+     * @param power
+     */
+    public void setLauncherPower(double power) {
+        masterMotor.ensureTalonControlMode(TalonControlMode.PercentVbus);
+        masterMotor.set(power);
+    }
+    
+    /**
+     * Returns the last set motor power. Only works if the motor is currently in PercentVbus mode - otherwise, returns 0.
+     * @return
+     */
+    public double getLauncherPower() {
+        if (masterMotor.getControlMode() != TalonControlMode.PercentVbus) {
+            return 0;
+        }
+        
+        double inversionFactor = masterMotor.getInverted() ? -1 : 1;
+        
+        return masterMotor.get() * inversionFactor;
+    }
+
+    /**
+     * Gives the shooter a new speed goal.
+     * @param shooterTargetSpeed in Rotations per Second (RPS)
+     */
+    public void setLauncherTargetSpeed(double speed) {
+        masterMotor.ensureTalonControlMode(TalonControlMode.Speed);
+        
+        // Update property for dashboard
+        shooterTargetSpeed.set(speed);
+        // Instruct motor about new speed goal
+        masterMotor.set(speed);
+        
+        // If there have been any changes to encoder settings or PID setings, apply them.
+        updateMotorConfig(masterMotor);
+    }
+    
+    public double getSpeed() {
+        return masterMotor.getSpeed();
+    }
+    
+    public boolean isAtSpeed() {
+        return Math.abs(getSpeed() - shooterTargetSpeed.get()) <= shooterSpeedThresh.get();
+    }
+    
+    public double getLauncherTargetSpeed() {
+        return shooterTargetSpeed.get();
+    }
+    
+    /**
+     * Updates driver station telemetry about the FuelLauncherSubsystem. Try to call this once per scheduler pass-through.
+     */
+    public void updateTelemetry() {
+        masterMotor.updateTelemetryProperties();        
+        atSpeedProp.set(isAtSpeed());
+        shooterCurrentSpeed.set(getSpeed());
+        shooterOutputPower.set(masterMotor.getOutputVoltage() / masterMotor.getBusVoltage());
+        shooterTalonError.set(masterMotor.getClosedLoopError());
+        
+        double currentTime = Timer.getFPGATimestamp();
+        // voltage, speed, error. What's the best ordering of these things? 
+        // voltage, error, speed. Ideally they would also be restricted to a constant number of digits
+        // to prevent problems.
+        
+        // In the short term, we can pull log files off the RIO for processing, though it adds in
+        // latency.
+        
+        // Format: time, voltage, error, speed
+        
+        log.info(currentTime + "," 
```

Comment:
This should be a Boolean property. I'd suggest making it ephemeral so that it is reset after a reboot. We only rarely need this, and this method is currently going to be run constantly because it's part of the default command.